src/command/archive/common.c

```diff
@@ -447,6 +447,49 @@ walSegmentNext(const String *const walSegment, const size_t walSegmentSize, cons
     FUNCTION_LOG_RETURN(STRING, strNewFmt("%08X%08X%08X", timeline, major, minor));
 }
 
+/**********************************************************************************************************************************/
+FN_EXTERN int
+walSegmentDist(const String *const walSegmentA, const String *const walSegmentB, const size_t walSegmentSize, const unsigned int pgVersion)
+{
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(STRING, walSegmentA);
+        FUNCTION_LOG_PARAM(STRING, walSegmentB);
+        FUNCTION_LOG_PARAM(SIZE, walSegmentSize);
+        FUNCTION_LOG_PARAM(UINT, pgVersion);
+    FUNCTION_LOG_END();
+
+    ASSERT(walSegmentA != NULL);
+    ASSERT(walSegmentB != NULL);
+    ASSERT(strSize(walSegmentA) == 24);
+    ASSERT(strSize(walSegmentB) == 24);
+    ASSERT(UINT32_MAX % walSegmentSize == walSegmentSize - 1);
```

Comment:
Consider adding a comment explaining why this assertion is necessary, as it may not be immediately clear to all readers.